### PR TITLE
ModerationCase entity

### PR DESCRIPTION
**Pull request by isoos**
- https://github.com/dart-lang/pub-dev/issues/7535
- not in use yet, just the implementation of the specification

### CODE REVIEW

File: app/lib/admin/models.dart [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  /// - `moderation-applied`, if this is a notification (kind = notification) and
+  ///   we decided to apply content moderation.
+  /// - `no-action-upheld`, if this is an appeal (kind = appeal) of a notification
+  ///   where we took no-action, and we decided to uphold that decision.
+  /// - `no-action-reverted`, if this is an appeal (kind = appeal) of a notification
+  ///   where we took no-action, and we decided to revert that decision.
+  /// - `moderation-upheld`, if this is an appeal (kind = appeal) of a notification
+  ///   where we applied content moderation, and we decided to uphold that decision.
+  /// - `moderation-reverted`, if this is an appeal (kind = appeal) of a notification
+  ///   where we applied content moderation, and we decided to revert that decision.
```
Comment (sigurdm) [inline app/lib/admin/models.dart:72]: Perhaps document the `null` case.

**Review by sigurdm — APPROVED** [review]

File: app/lib/admin/models.dart (hunk 1 above)
Comment (sigurdm) [reply app/lib/admin/models.dart:72]: SG

File: app/lib/admin/models.dart [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  late DateTime opened;
+
+  /// The case was resolved at this time (or null if it has not been resolved yet).
+  @db.DateTimeProperty()
+  DateTime? resolved;
+
+  /// The source of the case, one of:
+  /// - `external-notification`,
+  /// - `internal-notification` (only used for reports from @google.com accounts), or,
+  /// - `automated-detection. (will not be used)
```
Comment (jonasfj) [inline app/lib/admin/models.dart:36]:
```suggestion
  /// - `automated-detection`. (will not be used)
```

File: app/lib/admin/models.dart (hunk 1 above)
Comment (jonasfj) [reply app/lib/admin/models.dart:72]: IF we want a pending thing, we should rename it `status`.

For querying it's probably fine to have a `pending` value.

**Review by jonasfj — APPROVED** [review]

**Comment by isoos** [comment]
> IF we want a pending thing, we should rename it status.

Maybe it is better to reduce the possible values to `pending`, `no-action`, `upheld`, `reverted`, without the `${kind}` classification?

**Pull request merged**